### PR TITLE
test: add first gherkin proposal for test implementation

### PR DESCRIPTION
**Pull request by ghRocam**
The goal of this PR is to create the basic and esential tests that covers critical cases like users having access without full board info.

### CODE REVIEW

File: cypress/e2e/Transcript/Access/BoardAndDoBIsRequiredToSomeBoards.feature [hunk 1]
```diff
@@ -0,0 +1,16 @@
...
+
+  Scenario: user has empty board or date of birth details
+    Given an user did not fill the board details required by its board selection
+    When the user try to access the transcripts
+    Then a modal shows requiring to the user fill the missing board details
+    And the modal is blocking the transcript page behind
+
+  Scenarios: user is filling board or date of birth details
+    Given the modal to fill the board details is shown
+    When the user submit the missing board details
```
Comment (acabreragnz) [inline cypress/e2e/Transcript/Access/BoardAndDoBIsRequiredToSomeBoards.feature:15]:
```suggestion
    When the user submits the missing board details
```

File: cypress/e2e/Transcript/Access/BoardAndDoBIsRequiredToSomeBoards.feature [hunk 2]
```diff
@@ -0,0 +1,16 @@
+Feature: The user is required to complete their profile to see the transcripts
+  Background:
+    Given I am a registered user
+    And I have selected that I have a medical board
+    And I did not provide the required <MedicalBoardInformation>
+
+  Scenario: user has empty board or date of birth details
+    Given an user did not fill the board details required by its board selection
+    When the user try to access the transcripts
```
Comment (acabreragnz) [inline cypress/e2e/Transcript/Access/BoardAndDoBIsRequiredToSomeBoards.feature:9]:
```suggestion
    When the user tries to access the transcripts page
```

File: cypress/e2e/Transcript/Access/BoardAndDoBIsRequiredToSomeBoards.feature [hunk 3]
```diff
@@ -0,0 +1,16 @@
+Feature: The user is required to complete their profile to see the transcripts
+  Background:
+    Given I am a registered user
+    And I have selected that I have a medical board
+    And I did not provide the required <MedicalBoardInformation>
```
Comment (acabreragnz) [inline cypress/e2e/Transcript/Access/BoardAndDoBIsRequiredToSomeBoards.feature:5]: Please, don't use 1st person, to be consistent with the scenarios and other tests where we talk in the 3rd.

File: cypress/e2e/Transcript/Access/BoardAndDoBIsRequiredToSomeBoards.feature [hunk 4]
```diff
@@ -0,0 +1,16 @@
...
+
+  Scenario: user has empty board or date of birth details
```
Comment (acabreragnz) [inline cypress/e2e/Transcript/Access/BoardAndDoBIsRequiredToSomeBoards.feature:7]: We should mention what happens with the transcript access

File: cypress/e2e/Transcript/Access/BoardAndDoBIsRequiredToSomeBoards.feature [hunk 5]
```diff
@@ -0,0 +1,16 @@
...
+    Given an user did not fill the board details required by its board selection
+    When the user try to access the transcripts
+    Then a modal shows requiring to the user fill the missing board details
+    And the modal is blocking the transcript page behind
+
+  Scenarios: user is filling board or date of birth details
```
Comment (acabreragnz) [inline cypress/e2e/Transcript/Access/BoardAndDoBIsRequiredToSomeBoards.feature:13]: We should mention what happens with the transcript access (I know is in the feature)

File: cypress/e2e/Transcript/Access/ThereAreNoMissingBoardDetails.feature [hunk 6]
```diff
@@ -0,0 +1,10 @@
+Feature: The user can access the transcripts
+  Background:
+    Given I am a registered user
+    And I provided the required <MedicalBoardInformation> if any
+
+  Scenario: user has no missing board details
```
Comment (acabreragnz) [inline cypress/e2e/Transcript/Access/ThereAreNoMissingBoardDetails.feature:6]: We should mention what happens with the transcript access (I know is in the feature)

File: cypress/e2e/Transcript/Access/BoardAndDoBIsRequiredToSomeBoards.feature [hunk 7]
```diff
@@ -0,0 +1,16 @@
...
+    Given the modal to fill the board details is shown
+    When the user submit the missing board details
+    Then close the modal and the transcript page is not blocked
```
Comment (acabreragnz) [inline cypress/e2e/Transcript/Access/BoardAndDoBIsRequiredToSomeBoards.feature:16]:
```suggestion
    Then the modal is automatically closed and the transcript page is not blocked
```

File: cypress/e2e/Transcript/Access/BoardAndDoBIsRequiredToSomeBoards.feature [hunk 8]
```diff
@@ -0,0 +1,16 @@
+Feature: The user is required to complete their profile to see the transcripts
+  Background:
+    Given I am a registered user
+    And I have selected that I have a medical board
+    And I did not provide the required <MedicalBoardInformation>
+
+  Scenario: user has empty board or date of birth details
+    Given an user did not fill the board details required by its board selection
```
Comment (acabreragnz) [inline cypress/e2e/Transcript/Access/BoardAndDoBIsRequiredToSomeBoards.feature:8]: In general, we always write in present, but we have two options:

```
Given in past
When in present
Then in future
```

or

```
Given in present
When in present
Then in present
```

We have been using the last one for other tests

File: cypress/e2e/Transcript/Access/BoardAndDoBIsRequiredToSomeBoards.feature [hunk 9]
```diff
@@ -0,0 +1,16 @@
...
+    When the user try to access the transcripts
+    Then a modal shows requiring to the user fill the missing board details
```
Comment (acabreragnz) [inline cypress/e2e/Transcript/Access/BoardAndDoBIsRequiredToSomeBoards.feature:10]:
```suggestion
    Then a modal is shown requiring to the user fill the missing board details
```

**Review by acabreragnz — APPROVED** [review]
It looks great, just some comments (only change the ones you consider important)

File: cypress/e2e/Transcript/Access/ThereAreNoMissingBoardDetails.feature [hunk 10]
```diff
@@ -0,0 +1,10 @@
...
+    Given the user has complete board details in its profile
+    When the user attempt to access the transcripts
+    Then he can navigate to the transcript page
```
Comment (acabreragnz) [inline cypress/e2e/Transcript/Access/ThereAreNoMissingBoardDetails.feature:9]:
```suggestion
    Then the user should be navigated to the transcript page
```

Comment (acabreragnz) [reply cypress/e2e/Transcript/Access/ThereAreNoMissingBoardDetails.feature:9]: Remember to use `should` in the `Then`, look at the certificates examples.

File: cypress/e2e/Transcript/Access/BoardAndDoBIsRequiredToSomeBoards.feature (hunk 4 above)
Comment (ghRocam) [reply cypress/e2e/Transcript/Access/BoardAndDoBIsRequiredToSomeBoards.feature:7]: which line you mean?

File: cypress/e2e/Transcript/Access/BoardAndDoBIsRequiredToSomeBoards.feature (hunk 5 above)
Comment (ghRocam) [reply cypress/e2e/Transcript/Access/BoardAndDoBIsRequiredToSomeBoards.feature:13]: which line you mean?

File: cypress/e2e/Transcript/Access/ThereAreNoMissingBoardDetails.feature (hunk 6 above)
Comment (ghRocam) [reply cypress/e2e/Transcript/Access/ThereAreNoMissingBoardDetails.feature:6]: which line you mean? can you give an example please

File: cypress/e2e/Transcript/Access/BoardAndDoBIsRequiredToSomeBoards.feature (hunk 4 above)
Comment (acabreragnz) [reply cypress/e2e/Transcript/Access/BoardAndDoBIsRequiredToSomeBoards.feature:7]: Line 7, the scenario title.

`Scenario: user having empty board or date of birth details cannot access the transcript page`

File: cypress/e2e/Transcript/Access/BoardAndDoBIsRequiredToSomeBoards.feature (hunk 5 above)
Comment (acabreragnz) [reply cypress/e2e/Transcript/Access/BoardAndDoBIsRequiredToSomeBoards.feature:13]: Line 7, the scenario title.

File: cypress/e2e/Transcript/Access/ThereAreNoMissingBoardDetails.feature (hunk 6 above)
Comment (acabreragnz) [reply cypress/e2e/Transcript/Access/ThereAreNoMissingBoardDetails.feature:6]: Line 6, the scenario title.

File: cypress/e2e/Transcript/Access/BoardAndDoBIsRequiredToSomeBoards/BoardAndDoBIsRequiredToSomeBoards.js [hunk 11]
```diff
@@ -0,0 +1,78 @@
...
+
+Given('the modal is blocking the transcript page behind', () => {
+  cy.getByTestId('veil').should('be.visible');
+});
+
+Then('the modal is automatically closed and the transcript page is not blocked', () => {
+  cy.getByTestId('veil').should('not.be.visible');
+  cy.get('.aca-dialog-container').should('not.exist');
+
+  // we need to reset to the initial status this test after everything
```
Comment (acabreragnz) [inline cypress/e2e/Transcript/Access/BoardAndDoBIsRequiredToSomeBoards/BoardAndDoBIsRequiredToSomeBoards.js:62]: Please use a `before/beforeall` hook as commented here

https://github.com/bixlabs/CypressAcapediaProfessionals/pull/24

File: cypress/e2e/Transcript/Access/ThereAreNoMissingBoardDetails/ThereAreNoMissingBoardDetails.js [hunk 12]
```diff
@@ -0,0 +1,50 @@
...
+  cy.visit('/');
+});
+
+Given('they provided the required <MedicalBoardInformation> if any', () => {
+  cy.contains('Provide your medical board to report your MOC and CME credits.').should('not.exist');
+  cy.contains('Complete profile').should('not.exist');
+});
+
+Given('the user has complete board details in its profile', () => {
+  cy.get('body').then($body => {
```
Comment (acabreragnz) [inline cypress/e2e/Transcript/Access/ThereAreNoMissingBoardDetails/ThereAreNoMissingBoardDetails.js:16]: Can you add a comment explaining this condition?

**Review by acabreragnz — APPROVED** [review]
LGTM, a small change to use cucumber hooks instead.

**Pull request merged**